frontend/src/pages/Categories/Quiz/index.tsx

```diff
@@ -132,7 +133,8 @@ const QuizPage = ({ currentLogin: { user, token } }: Props) => {
 
   useEffect(() => {
     if (!categorySlug) return;
-    if (!choices || !userProgress) return;
+    if (!choices) return;
+    if (!userProgress) return;
```

Comment:
refactor this
eg: `if (!category || !questions || !answers || !choices) return;`